src/coreclr/vm/weakreferencenative.cpp

```diff
@@ -139,13 +139,15 @@ NativeComWeakHandleInfo* GetComWeakReferenceInfo(OBJECTREF* pObject)
     // If the object is a managed type deriving from a COM type, then we also do not want to use a native COM
     // weak reference to it.  (Otherwise, we'll wind up resolving IWeakReference-s back into the CLR
     // when we don't want to have reentrancy).
+#ifdef FEATURE_COMINTEROP
     if (pMT->IsComObjectType()
      && (pMT == g_pBaseCOMObject || !pMT->IsExtensibleRCW()))
     {
         pWeakReferenceSource = reinterpret_cast<IWeakReferenceSource*>(GetComIPFromObjectRef(pObject, IID_IWeakReferenceSource, false /* throwIfNoComIP */));
     }
-#ifdef FEATURE_COMWRAPPERS
     else
```

Comment:
This is a bit dangerous. The issue becomes if `FEATURE_COMINTEROP` is set but `FEATURE_COMWRAPPERS` isn't then the below `if` statement on line 156 becomes and `else if` which is clearly incorrect. This might be okay because at present `FEATURE_COMINTEROP` will never be set by itself but that is something we might want to prevent.

Reply:
Ah, yeah, we should fix that. I think just moving `#ifdef FEATURE_COMWRAPPERS`/`#endif` below to be inside the braces  (so that if it isn't defined we just have an empty else block) is probably the simplest way.

Reply:
Ok